aptos-move/framework/aptos-framework/sources/Coin.move

```diff
@@ -29,6 +29,7 @@ module AptosFramework::Coin {
     struct CoinStore<phantom CoinType> has key {
         coin: Coin<CoinType>,
         deposit_events: EventHandle<DepositEvent>,
+        register_events: EventHandle<RegisterEvent>,
```

Comment:
won't this only be accessible if we know the CoinType? Your demo suggests yes. Make a resource distinct for all coins.

Also Make it address::MoonCoin::MoonCoin or something

Reply:
Updated. I added a separate untyped CoinEvents (better name suggestions are welcomed) that are added to each account when register is called. We can also later add initialize_events in this same struct as well when an account initializes their coin. Also updated the demo/screenshot.